UIKitTests/UIScrollViewTests.swift

```diff
@@ -47,6 +47,65 @@ class UIScrollViewTests: XCTestCase {
         XCTAssertEqual(scrollView.contentInset, arbitraryContentInset)
     }
 
+    func testScrollIndicatorsViewHierarchyPositions() {
+        // subviews[0] is the back-most view, higher indexes are higher in layer hierarchy
+        // we want scroll indicators to always remain on top of 'normal' subviews
+        // and we want newer 'normal' subviews to be above older ones
+
+        // TODO: iOS does not allow explicit access to scroll indicators via subviews
+        // we should remove that, too - but for now it's the easy way to go and helpful in testing
+
+        let view1 = UIView()
+        let view2 = UIView()
+        let view3 = UIView()
+
+        scrollView.addSubview(view1)
+        scrollView.addSubview(view2)
+        scrollView.addSubview(view3)
+
+        let horizontalIndicatorIndex = scrollView.subviews.index(of: scrollView.horizontalScrollIndicator)!
+        let verticalIndicatorIndex = scrollView.subviews.index(of: scrollView.verticalScrollIndicator)!
+        let label1Index = scrollView.subviews.index(of: view1)!
+        let label2Index = scrollView.subviews.index(of: view2)!
+        let label3Index = scrollView.subviews.index(of: view3)!
+
+        XCTAssert(horizontalIndicatorIndex > label1Index)
+        XCTAssert(verticalIndicatorIndex > label1Index)
+
+        XCTAssert(label2Index > label1Index)
+
+        XCTAssertEqual(label1Index, 0)
+        XCTAssertEqual(label2Index, 1)
+        XCTAssertEqual(label3Index, 2)
+        XCTAssertEqual(horizontalIndicatorIndex, 3)
+        XCTAssertEqual(verticalIndicatorIndex, 4)
+        XCTAssertEqual(scrollView.subviews.count, 5)
+
+        let viewToBeInserted1 = UIView()
+        scrollView.insertSubview(viewToBeInserted1, at: 1)
+        let indexOfInsertedView1 = scrollView.subviews.index(of: viewToBeInserted1)!
+
+        // This might seem weird, but that's what happens in iOS:
+        // the inserted view and the one that was in its place before will have the same index
```

Comment:
is it possible this is only weird because of how the test is written?

It doesn't seem that weird to me that the indexes will overlap once we insert a new subview if we're not getting the _new_ indexes after inserting the subview?

What am I missing here? It seems like we should get the _new_ indexes here which would be less confusing?